### ISSUE TITLE
feature request: --make ttf/cff/cff2/svg/blahblahblah

### ISSUE THREAD

**Issue by Pomax** [fonttools/fonttools#1393]
Every time I had a dumb or possibly-not-dumb-but-definitely-crazy idea for a font, I reach for TTX with a small font file and then strip down the converted XML to then start hacking on the initial font in XML form... but it would be super great if I didn't have to find a random small font to strip down, but instead had a way to say something like: `ttx --make-ttf` or `ttx --make cff2` and have it output minimal XML for a stub font with a null and space glyph with a 3/1/0 cmap or something, with a way to indicate whether you need TTF's loca+glyf, or a CFF/CFF2 block, or an SVG table, or even EBDT and friends?

There could even just be minimal ttf/cff/cff2/svg/ebdt fonts that are used to generate the XML by running "ttx thatfont.otf" under the hood or something. Stub fonts are truly tiny, so including all of them would pack maybe 2kb to the fonttools install?

Anyway, just a thought.

**Comment by behdad** [comment]
Yes please!!!  Now that there's fontBuilder module, maybe this becomes the command-line interface to that module...

**Comment by Pomax** [comment]
Ooh, is there fontbuilder specific documentation? I wouldn't mind seeing if I can stub something out for this.

**Comment by justvanrossum** [comment]
There are some examples in the source: https://github.com/fonttools/fonttools/blob/master/Lib/fontTools/fontBuilder.py
It's pretty new, and still requires quite some spec knowledge, but it does make it a lot easier to programmatically get a working font.

**Comment by Pomax** [comment]
Thanks! Spec knowledge should hopefully not be a problem here, given my past font parser and generator work =D

**Comment by behdad** [comment]
If someone wants to contribute such stub fonts, we can make this work. Otherwise not planned currently.

**Comment by Pomax** [comment]
might be worth keeping open but instead using the github projects system and putting it in an "icebox" or something with the "help wanted" and maybe even "good first bug" label?

**Comment by Pomax** [comment]
Come to think of it, fonttools might be a decent google summer of code project?
